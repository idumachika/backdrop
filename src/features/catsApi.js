import {createApi, fetchBaseQuery} from '@reduxjs/toolkit/query/react';

export const catsApi = createApi({
  reducerPath: 'catsApi',
  baseQuery: fetchBaseQuery({baseUrl: 'https://api.thecatapi.com/v1/'}),
  endpoints: builder => ({
    getAllCats: builder.query({
      query: () => 'breeds?limit=20',
    }),
  }),
});

export const {useGetAllCatsQuery} = catsApi;
