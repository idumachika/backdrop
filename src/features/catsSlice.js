import {createSlice} from '@reduxjs/toolkit';

export const catSlice = createSlice({
  name: 'cat',
  initialState: {
    favourite: false,
    cats: [],
  },
  reducers: {
    add(state, action) {
      const catExists = state.cats.find(item => item.id === action.payload.id);
      if (catExists) {
        return;
      }

      state.cats.push(action.payload);
    },
    remove(state, action) {
      state.cats = state.cats.filter(item => item.id !== action.payload);
    },
  },
});

export const {add, remove} = catSlice.actions;
export default catSlice.reducer;
