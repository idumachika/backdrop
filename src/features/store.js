import AsyncStorage from '@react-native-async-storage/async-storage';
import {configureStore} from '@reduxjs/toolkit';
import {persistReducer, persistStore} from 'redux-persist';
import catReducer from '../features/catsSlice';
import {catsApi} from './catsApi';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

export const store = configureStore({
  reducer: {
    [catsApi.reducerPath]: catsApi.reducer,
    cats: persistReducer(persistConfig, catReducer),
  },
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({serializableCheck: false}).concat(catsApi.middleware),
});

export const persistor = persistStore(store);
