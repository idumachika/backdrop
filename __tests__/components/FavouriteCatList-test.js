import React from 'react';
import { Provider } from 'react-redux';
import { persistor, store } from '../../src/features/store';
import { fireEvent, render } from '@testing-library/react-native';
import FavouriteCatlist from '../../src/components/FavouriteCatList';
import { PersistGate } from 'redux-persist/integration/react';


describe('<FavouriteCatlist/> tests', () => {
    test('should render the FavouriteCatlist', () => {
        const { toJSON } =
            render(<Provider store={store}>
                <PersistGate persistor={persistor}>
                    <FavouriteCatlist />
                </PersistGate>
            </Provider>);
        expect(toJSON()).toMatchSnapshot();


    });

    // it('should fire onPress when pressed', () => {
    //     const onPressMock = jest.fn();
    //     const { queryByTestId } = render(<CryptoCard onPress={onPressMock} />);
    //     fireEvent.press(queryByTestId('crypto-card'));

    //     expect(onPressMock).toHaveBeenCalledTimes(1);
    // });

    // it('should display a title (name), amount, amountInBtc, timeFrame', () => {
    //     const { getByText } = render(
    //         <CryptoCard
    //             name="Btc"
    //         // amount="2,000"
    //         // amountInBtc="0.002"
    //         // timeFrame="24 hrs"
    //         />,
    //     );

    //     expect(getByText('Btc')).toBeTruthy();
    //     // expect(getByText('2,000')).toBeTruthy();
    //     // expect(getByText('0.002')).toBeTruthy();
    //     // expect(getByText('24 hrs')).toBeTruthy();
    // });
});
