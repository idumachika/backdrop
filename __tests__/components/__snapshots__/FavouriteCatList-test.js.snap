// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavouriteCatlist/> tests should render the FavouriteCatlist 1`] = `null`;
